pkg/server/conn.go

```diff
@@ -801,6 +810,11 @@ func (cc *clientConn) openSessionAndDoAuth(authData []byte, authPlugin string, z
 		return err
 	}
 
+	err = cc.checkUserConnectionCount(host)
```

Comment:
this check happens before `incrementUserConnectionsCounter`, the value maybe be changed after the check and before the increment, so limitation may not take effect. How about let `incrementUserConnectionsCounter` return the error of limit exceeded?

Reply:
Midified it. 
Check and increase the count of connections when call incrementUserConnectionsCounter().